clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/securitygroup/SecurityGroupLookupFactory.groovy

```diff
@@ -67,14 +68,26 @@ class SecurityGroupLookupFactory {
       this.accounts = accounts
     }
 
-    private NetflixAmazonCredentials getCredentialsForName(String accountName) {
+    NetflixAmazonCredentials getCredentialsForName(String accountName) {
       accounts.find { it.name == accountName }
     }
 
+    NetflixAmazonCredentials getCredentialsForId(String accountId) {
+      accounts.find { it.accountId == accountId }
+    }
+
     String getAccountIdForName(String accountName) {
       getCredentialsForName(accountName)?.accountId ?: accountName
     }
 
+    String getAccountNameForId(String accountId) {
+      accounts.find { it.accountId == accountId }?.name ?: accountId
+    }
+
+    boolean accountIdExists(String accountId) {
+      accounts.any { it.accountId == accountId }
+    }
+
     SecurityGroupUpdater createSecurityGroup(UpsertSecurityGroupDescription description) {
```

Comment:
Might make sense to make these return `Optional<SecurityGroupUpdater>` - most of the usages are immediately doing a null check after they get a `SecurityGroupUpdater` back from this guy